src/main/java/org/javarosa/core/model/condition/Condition.java

```diff
@@ -58,12 +58,8 @@ public Condition() {
 
     }
 
-    public Condition(IConditionExpr expr, int trueAction, int falseAction, TreeReference contextRef) {
-        this(expr, trueAction, falseAction, contextRef, new ArrayList<>(0));
-    }
-
-    public Condition(IConditionExpr expr, int trueAction, int falseAction, TreeReference contextRef, ArrayList<TreeReference> targets) {
-        super(expr, contextRef, targets);
+    protected Condition(IConditionExpr expr, TreeReference contextRef, TreeReference originalContextRef, List<TreeReference> targets, Set<QuickTriggerable> immediateCascades, int trueAction, int falseAction) {
```

Comment:
Not clear on implications of adding `originalContextRef and `immediateCascades` here. Seems `immediateCascades` may only have been needed for fast 2014?

Reply:
This should be safe. I re-routed Triggerable creation to a couple of factories inside it. this only makes explicit all the parts we need to build one.